config/config/src/main/java/io/helidon/config/ConfigMapperManager.java

```diff
@@ -161,6 +161,7 @@ static final class MapperProviders {
                 Optional<? extends BiFunction<Config, ConfigMapper, ?>>>> providers = new LinkedList<>();
 
         private MapperProviders() {
+            add(new EnumMapperProvider());
```

Comment:
I would actually prefer to add the `EnumMapperProvider` through service loader as any other, with a very low priority (e.g. Priority 10000 in Helidon 3.x and 2.x, Weight 0 in Helidon 4).
This will give it the same treatment as any user created provider (and an option to use Helidon service loader features for it)

Reply:
There are two way to do this:
1. Add `provides` to `module-info.java` (needs to be public)
2. Add it manually in `BuilderImpl.loadMapperServices(List)` to the `HelidonServiceLoader.Builder` (can be package local)

Reply:
I can change this easily and, yes, I have supplied service loader impls in several places before 😉 

My original thinking was that this is _not_ a user-provided mapper provider but the first Helidon-provided one (unless I missed something) and therefore it made sense to include it directly as we do with so many of the built-in mappers.

Either way is fine with me based on your preference. If I change it to be loaded as a service I'll not use the `provides` approach. No need for the provider to be public.

Reply:
Changed

Reply:
I'm missing the basic premise as to why we feel we need a pluggable manager for dealing with Enums.  It can be implemented much more simpler in the base Config classes using the code snippet I provided.  Can someone explain why we'd want to have a manager for dealing with Enum type A that is different from a manager for dealing with Enum type B?  Or why these need to be overridable?  I still feel we should treat Enums as a primitive type, and built into the core config library.

Reply:
If the config system were not already written the way it is, there might not be a compelling reason to _create_ a way for the enum conversion to be pluggable and overridable. But given the code as it stands, creating a mapper provider for enum conversion is the right approach (see below). Tomas's suggestion for how to incorporate our enum mapper provider reuses existing code which happens to make it pluggable and overridable. Developers might never write their own but doing it this way follows the existing patterns and conventions in the config subsystem for providing conversion rather than inventing some new mechanism.

There is no single "base `Config#as`" method, as your earlier suggestion assumed, where your three lines might be inserted. I pointed before to what the `as` method looks like in one implementation of the `Config` interface.
If you look at the logic which starts there, you see that the code there does _not_  perform the conversion. 

Instead--and I'm simplifying a bit--that code creates a `ConfigValue` instance and stores with it a function which, given the type passed to `as`, will choose the mapper to apply at the _later_ point when conversion _does_ occur.

Only later, when the converted value is actually needed (e.g., when `ifPresent` or `get` are invoked), does config invoke the saved function to choose the mapper and then invoke the mapper to perform the conversion.

That's why we need a way--that fits in with the way config is designed--of delivering enum mappers.

I've explained before why we need different mappers for different specific enum types, and therefore why we need a mapper provider rather than a single mapper for all enums, but to restate: 

Config's mapper selection is by the type that was stored with the `ConfigValue` instance created by `as`. So when the developer invokes `config.as(Color.class)` then `Color` is the type--not `Enum`--config will later use to find a mapper. So we'd better be prepared to deliver config a mapper for `Color` when config asks for one.

The new enum mapper provider does exactly that--it creates a mapper on-demand that converts a string to `Color` or `Shape` or whatever enum type might arise.